src/features/import-map.js

```diff
@@ -44,6 +44,45 @@ systemJSPrototype.resolve = function (id, parentUrl) {
   return resolveImportMap(importMap, resolveIfNotPlainOrUrl(id, parentUrl) || id, parentUrl) || throwUnresolved(id, parentUrl);
 };
 
+let supportsPreload = false, supportsPrefetch = false;
+if (hasDocument) {
+  let relList = document.createElement('link').relList;
+  if (relList && relList.supports) {
+    supportsPrefetch = true;
+    try {
+      supportsPreload = relList.supports('preload');
+    }
+    catch (e) {}
+  }
+}
+
+const systemInstantiate = systemJSPrototype.instantiate;
+systemJSPrototype.instantiate = function (url, firstParentUrl) {
+  const depcache = importMap.depcache[url];
```

Comment:
Shouldn't this be done recursively for the whole dependency tree? Otherwise, we're just halving the waterfall effect, no?

Reply:
This PR matches [Guy's proposal to the import map spec](https://github.com/WICG/import-maps/blob/2bef7a5ee3de86a31b15abf074cf57c1b0019524/spec.bs#L479) 

I agree with his proposed spec - eagerly prefetching all of the modules in the import map is fairly irresponsible use of the network. It would result in a lot of unnecessary requests, which is [stated in the proposal as something to avoid](https://github.com/WICG/import-maps/blob/2bef7a5ee3de86a31b15abf074cf57c1b0019524/spec.bs#L481).